src/app/components-link.directive.ts

```diff
@@ -0,0 +1,39 @@
+import { Directive, ElementRef, Input, OnDestroy, OnInit } from '@angular/core';
+import { combineLatest, Subject } from 'rxjs';
+import { map, takeUntil } from 'rxjs/operators';
+import { ComponentsListSearchService } from './components-list.service';
+
+@Directive({
+  selector: '[npgComponentLink]',
+})
+export class ComponentLinkDirective implements OnInit, OnDestroy {
+  constructor(private el: ElementRef, private componentsListSearchService: ComponentsListSearchService) {}
+
+  @Input() npgComponentLink: string | any[] = '';
+  private destroy$ = new Subject<void>();
+
+  public ngOnInit(): void {
+    combineLatest([
+      this.componentsListSearchService.flatFilteredComponentLinkList$,
+      this.componentsListSearchService.activeElementIndex$,
+    ])
+      .pipe(
+        map(([flatArray, index]) => {
+          const componentLink = flatArray[index]?.link;
+          this.componentsListSearchService.selectedElement$.next(componentLink);
+          if (this.npgComponentLink === componentLink) {
+            this.el.nativeElement.classList.add('focus');
```

Comment:
Use Angular API to access DOM API, e.g. modify classes. It looks like it could be refactored to use `HostBinding` to `focus` class.